src/RootSolvers.jl

```diff
@@ -328,4 +354,233 @@ function find_zero(
     )
 end
 
+function find_zero(
+    f::F,
+    x0::FT,
+    x1::FT,
+    ::BisectionMethod,
+    soltype::SolutionType,
+    xatol::FT = FT(1 // 1000),
+    maxiters = ceil(Int, log2(abs(x1 - x0) / xatol)),
+) where {F, FT <: AbstractFloat}
+    y0, y1 = f(x0), f(x1)
+
+    x_history = init_history(soltype, FT)
+    push_history!(x_history, x0, soltype)
+    push_history!(x_history, x1, soltype)
+
+    y_history = init_history(soltype, FT)
+    push_history!(y_history, y0, soltype)
+    push_history!(y_history, y1, soltype)
+
+    if y0 * y1 > 0
+        return SolutionResults(soltype, x0, false, y0, 0, x_history, y_history)
+    end
+
+    for i in 1:maxiters
+        x2 = (x0 + x1) / 2
+        y2 = f(x2)
+        if y2 * y0 < 0
+            x1, y1 = x2, y2
+        else
+            x0, y0 = x2, y2
+        end
+        push_history!(x_history, x2, soltype)
+        push_history!(y_history, y2, soltype)
+    end
+    x = (x0 + x1) / 2
+    return SolutionResults(
+        soltype,
+        (x0 + x1) / 2,
+        true,
```

Comment:
Shouldn't we have convergence check inside the iteration? Is this method guaranteed to converge after `maxiter`? Also, what if `maxiter` is passed and is not equal to the default?

Reply:
Bisection is guaranteed to converge to a given bracket size in a given number of iterations. I'll update all this with your new functor stuff soon.

Reply:
But you are right, that if maxiter is not the default there is no guaranteed convergence